Src/library/net/Clib/network.c

```diff
@@ -812,93 +829,180 @@ void c_put_float(EIF_INTEGER fd, EIF_REAL f)
 void c_put_double(EIF_INTEGER fd, EIF_DOUBLE d)
 	/*x transmission of double d through socket fd */
 {
+	/* Unused (!) by EiffelNet */
 	double dbl;
 	dbl = ise_htond(d);
 	CPUTX(fd,(char *) &dbl,sizeof(dbl))
 }
 
+EIF_INTEGER c_put_stream_noexception(EIF_INTEGER fd, EIF_POINTER stream_pointer, EIF_INTEGER length)
+	/* transmission of string s of size size trought socket fd.
+	 *	NO exception is raised, and eventual error is return as result!
+	 */
+{
+	int res;
+	CSENDX(res, fd, stream_pointer, length, 0);
+	return (EIF_INTEGER) res;
+}
+
 void c_put_stream(EIF_INTEGER fd, EIF_POINTER stream_pointer, EIF_INTEGER length)
 	/*x transmission of string s of size size trought socket fd */
 {
 	CPUTX(fd, (char *) stream_pointer,length)
```

Comment:
Use CSENDX macro instead with an explicit call to eif_net_check. That will be more clear that c_put_stream and c_put_stream_noexception have similar implementation and only differs by the call to eif_net_check. This will be similar to the new c_read_stream and c_read_stream_noexception implementations